valkyrie/lib/valkyrie/specs/shared_specs/file.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+RSpec.shared_examples 'a Valkyrie::StorageAdapter::File' do
+  subject(:file) { described_class.new(id: "test_file", io: io) }
+  let(:io) { instance_double(::File) }
+
+  describe '#io' do
+    it 'returns the IO Object' do
+      expect(file.io).to eq io
+    end
+  end
+
+  describe '#read' do
+    before do
+      allow(io).to receive(:read).and_return('Lorem ipsum dolor sit amet, consectetur adipiscing elit.')
+    end
+    it 'reads from the IO Object' do
```

Comment:
Something as simple as `respond_to :read` should be good enough for the shared spec.